Xamarin.Forms.Platform.UAP/Dispatcher.cs

```diff
@@ -25,6 +25,6 @@ public Dispatcher(CoreDispatcher coreDispatcher)
 			_coreDispatcher = coreDispatcher;
 		}
 
-		bool IDispatcher.IsInvokeRequired => Device.IsInvokeRequired;
+		bool IDispatcher.IsInvokeRequired => !_coreDispatcher.HasThreadAccess;
```

Comment:
This is the main fix to use the correct core dispatcher. The `Device.IsInvokeRequired` implementation only verifies against the main/first dispatcher.